velox/tool/trace/OperatorReplayerBase.cpp

```diff
@@ -31,17 +33,46 @@ DEFINE_string(
     task_id,
     "",
     "Specify the target task id, if empty, show the summary of all the traced query task.");
+DEFINE_string(node_id, "", "Specify the target node id.");
+DEFINE_int32(pipeline_id, 0, "Specify the target pipeline id.");
+DEFINE_string(operator_type, "", "Specify the target operator type.");
+DEFINE_string(
+    table_writer_output_dir,
+    "",
+    "Specify output directory of TableWriter.");
 
 using namespace facebook::velox;
 
 namespace facebook::velox::tool::trace {
 
-QueryTraceReplayer::QueryTraceReplayer()
-    : rootDir_(FLAGS_root), taskId_(FLAGS_task_id) {}
+OperatorReplayerBase::OperatorReplayerBase(
+    std::string rootDir,
+    std::string taskId,
+    std::string nodeId,
+    int32_t pipelineId,
+    std::string operatorType)
+    : rootDir_(std::move(rootDir)),
+      taskId_(std::move(taskId)),
+      nodeId_(std::move(nodeId)),
+      pipelineId_(pipelineId),
+      operatorType_(std::move(operatorType)) {
+  VELOX_USER_CHECK(!rootDir_.empty());
+  VELOX_USER_CHECK(!taskId_.empty());
+  VELOX_USER_CHECK(!nodeId_.empty());
+  VELOX_USER_CHECK_GE(pipelineId_, 0);
+  VELOX_USER_CHECK(!operatorType_.empty());
+  const auto traceTaskDir = fmt::format("{}/{}", rootDir_, taskId_);
+  const auto metadataReader = exec::trace::QueryMetadataReader(
+      traceTaskDir, memory::MemoryManager::getInstance()->tracePool());
+  metadataReader.read(queryConfigs_, connectorConfigs_, planFragment_);
+  queryConfigs_[core::QueryConfig::kQueryTraceEnabled] = "false";
+  fs_ = filesystems::getFileSystem(rootDir_, nullptr);
+  maxDrivers_ = exec::trace::getNumDrivers(
+      fmt::format("{}/{}/{}", traceTaskDir, nodeId_, pipelineId_), fs_);
```

Comment:
Shall we construct the actual path for pipeline id inside getNumDrivers? getNumDrivers it is better to take the root dir and better to pass task_id to utility to ease use?